File: mdetect/src/main/java/me/myatminsoe/mdetect/MDetect.java
package me.myatminsoe.mdetect;

import android.content.Context;
import android.util.Log;
import android.view.ViewGroup;
import android.widget.TextView;

public class MDetect {

    private static final int TYPE_UNICODE = 0;
    private static final int TYPE_ZAWGYI = 1;
    private static final int TYPE_DEFAULT = 2;

    private static int isUnicode = TYPE_DEFAULT;

    private MDetect() {
    }

    public static void init(Context context) {
        if(isUnicode != TYPE_DEFAULT)
            return;
        TextView tv = new TextView(context, null);
        tv.setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT,
                ViewGroup.LayoutParams.WRAP_CONTENT));

        tv.setText("\u1000");
        tv.measure(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        int length1 = tv.getMeasuredWidth();

        tv.setText("\u1000\u1039\u1000");
        tv.measure(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        int length2 = tv.getMeasuredWidth();

        Log.i("MDetect", length1 + ", " + length2);

        isUnicode = length1 == length2 ? TYPE_UNICODE : TYPE_ZAWGYI;
    }

    /**
     * method for getting user's encoding
     *
     * @return whether user's encoding follows myanmar unicode standard
     */
    public static boolean isUnicode() {
        if (isUnicode == TYPE_DEFAULT)
            throw new UnsupportedOperationException("MDetect was not initialized.");
        return isUnicode == TYPE_UNICODE;
    }

    public static String getString(String unicodeString) {
        if (isUnicode())
            return unicodeString;
        return Rabbit.uni2zg(unicodeString);
    }

    public static String[] getStringArray(String[] unicodeStringArray) {
        if (isUnicode())
            return unicodeStringArray;
        for (int i = 0; i < unicodeStringArray.length; i++) {
            unicodeStringArray[i] = Rabbit.uni2zg(unicodeStringArray[i]);
        }
        return unicodeStringArray;
    }
}

File: mdetect/src/main/java/me/myatminsoe/mdetect/MDetect.kt
package me.myatminsoe.mdetect

import android.content.Context
import android.util.Log
import android.view.ViewGroup
import android.widget.TextView
import com.comquas.rabbit.Rabbit

object MDetect {

    private var cacheUnicode: Boolean? = null


    fun init(context: Context) {
        if (cacheUnicode != null) {
            Log.i("MDetect", "MDetect was already initialized.")
            return
        }
        val textView = TextView(context, null)
        textView.layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT,
                ViewGroup.LayoutParams.WRAP_CONTENT)

        textView.text = "\u1000"
        textView.measure(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        val length1 = textView.measuredWidth

        textView.text = "\u1000\u1039\u1000"
        textView.measure(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        val length2 = textView.measuredWidth

        cacheUnicode = length1 == length2
    }

    /**
     * method for getting user's encoding

     * @return whether the device follows myanmar unicode standard
     */
    fun isUnicode(): Boolean {
        if (null == cacheUnicode)
            throw UnsupportedOperationException("MDetect was not initialized.")
        return cacheUnicode!!
    }

    /**
     * method for getting display text
     * @param unicodeString Unicode String
     * @return appropriate String according to device's encoding
     */
    fun getText(unicodeString: String): String {
        return if (isUnicode()) unicodeString else Rabbit.uni2zg(unicodeString)
    }

    /**
     * method for getting String from user input
     * @param inputString String inputted by user
     * @return Unicode String
     */
    fun getInputText(inputString: String): String {
        return if (isUnicode()) inputString else Rabbit.zg2uni(inputString)
    }

    /**
     * method for getting string array
     * @param unicodeStringArray Unicode String Array
     * @return appropriate String Array according to device's encoding
     */
    fun getStringArray(unicodeStringArray: Array<String>): Array<String> {
        if (isUnicode())
            return unicodeStringArray
        val zawgyiStringArray = unicodeStringArray.clone()
        for (i in zawgyiStringArray.indices) {
            zawgyiStringArray[i] = Rabbit.uni2zg(zawgyiStringArray[i])
        }
        return zawgyiStringArray
    }
}
